### PR TITLE
DAOS-6659 continue renaming of daos_io_server to daos_engine

### PR DESCRIPTION
**Pull request by Michael-Hennecke**
rename daemon logfiles to daos_server.log and daos_engine.N.log
where needed, change server to engine in config file comments

Signed-off-by: Michael Hennecke <mhennecke@lenovo.com>

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]
thanks for this Michael

**Review by tanabarr — CHANGES_REQUESTED** [review]
actually while you're at it, would you mind updating "per-server" to "per-engine" in the config files as well please?

**Comment by tanabarr** [comment]
Could also add "Doc-only: true" to the commit message, not sure if this is being auto detected correctly.

**Review by johannlombardi — DISMISSED** [review]

File: utils/config/daos_server.yml [hunk 1]
```diff
@@ -290,7 +290,7 @@
 #  bdev_list: ["0000:5d:05.5]
 
 #-
-#  # Rank to be assigned as identifier for server.
+#  # Rank to be assigned as identifier for this engine.
```
Comment (jbrosenz) [inline utils/config/daos_server.yml:293]: @johannlombardi @mjmac is it time to change `rank` to `instance` or something similar now, given that this isn't actually the unique rank?

**Review by jbrosenz — COMMENTED** [review]
LGTM, pending the one question about 'rank'.

**Comment by tanabarr** [comment]
> actually while you're at it, would you mind updating "per-server" to "per-engine" in the config files as well please?

ignore that request I will do as part of a separate PR

**Review by tanabarr — DISMISSED** [review]

**Comment by Michael-Hennecke** [comment]
can somebody send me the details of the unit test failure? can't see the build logs from here...

**Comment by jolivier23** [comment]
Error Message
Failed
Stacktrace
=== RUN   TestServerConfig_Constructed
    server_test.go:275: (-want, +got):   &config.Server{
          	ControlPort:     10001,
          	TransportConfig: &{CertificateConfig: {ServerName: "server", ClientCertDir: "/etc/daos/certs/clients", CARootPath: "/etc/daos/certs/daosCA.crt", CertificatePath: "/etc/daos/certs/server.crt", ...}},
          	Servers:         nil,
          	Engines: []*engine.Config{
          		&{
          			... // 6 identical fields
          			SocketDir: "./.daos/daos_server",
          			LogMask:   "WARN",
        - 			LogFile:   "/tmp/daos_engine.0.log",
        + 			LogFile:   "/tmp/daos_server1.log",
          			Storage:   {SCM: {MountPoint: "/mnt/daos/1", Class: s"ram", RamdiskSize: 16}, Bdev: {Class: s"nvme", DeviceList: {"0000:81:00.0"}}},
          			Fabric:    {Provider: "ofi+verbs;ofi_rxm", Interface: "qib0", InterfacePort: 20000, PinnedNumaNode: &0, ...},
          			... // 2 identical fields
          		},
          		&{
          			... // 6 identical fields
          			SocketDir: "./.daos/daos_server",
          			LogMask:   "WARN",
        - 			LogFile:   "/tmp/daos_engine.1.log",
        + 			LogFile:   "/tmp/daos_server2.log",
          			Storage:   {SCM: {MountPoint: "/mnt/daos/2", Class: s"dcpm", DeviceList: {"/dev/pmem0"}}, Bdev: {Class: s"malloc", DeviceList: {"/tmp/daos-bdev1", "/tmp/daos-bdev2"}, DeviceCount: 1, FileSize: 4, ...}},
          			Fabric:    {Provider: "ofi+verbs;ofi_rxm", Interface: "qib1", InterfacePort: 20000, PinnedNumaNode: &1, ...},
          			... // 2 identical fields
          		},
          	},
          	BdevInclude: {"0000:81:00.1", "0000:81:00.2", "0000:81:00.3"},
          	BdevExclude: {"0000:81:00.1"},
          	... // 3 identical fields
          	SetHugepages:   false,
          	ControlLogMask: s"ERROR",
        - 	ControlLogFile: "/tmp/daos_server.log",
        + 	ControlLogFile: "/tmp/daos_control.log",
          	ControlLogJSON: false,
          	HelperLogFile:  "/tmp/daos_admin.log",
          	... // 3 ignored and 11 identical fields
          }
--- FAIL: TestServerConfig_Constructed (0.01s)

**Comment by jolivier23** [comment]
You should be able to run the tests locally

RUN_TEST_FILTER=go utils/run_test.sh

**Comment by Michael-Hennecke** [comment]
fixed the unit test in server_test.go to reflect changed logfile names

**Review by jolivier23 — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

File: utils/config/daos_server.yml (hunk 1 above)
Comment (mjmac) [reply utils/config/daos_server.yml:293]: This is in fact a rank in the system sense. Normally the ranks are assigned by the MS, but we do support allowing the rank to be set in the config.

File: utils/config/daos_server.yml [hunk 2]
```diff
@@ -304,7 +304,7 @@
 #
 #  targets: 20
 #
-#  # Number of helper XStreams per VOS server.
+#  # Number of helper XStreams per VOS engine.
```
Comment (mjmac) [inline utils/config/daos_server.yml:307]: Not sure if this particular substitution is correct. @jolivier23: Does this new terminology make sense? That having been said, I'm not sure that "VOS server" made sense either.

**Review by mjmac — DISMISSED** [review]
LGTM. If the "VOS engine" rename doesn't make sense we can fix that later, perhaps with something more descriptive.

File: utils/config/daos_server.yml (hunk 2 above)
Comment (johannlombardi) [reply utils/config/daos_server.yml:307]: Right, there is no such thing as VOS server. The original text did not make sense.
That's the number of I/O service threads (and network endpoints) per engine.

**Comment by johannlombardi** [comment]
I am fine to land it and fix the comment in a subsequent "doc-only" patch. My only worry is whether we need to change as well the paths/file names of the logs collected by the CI or if we are good. @mjmac @brianjmurrell @bfaccini  could you please advise?

**Comment by Michael-Hennecke** [comment]
The only place in CI that I could find is this:

ci/provisioning/log_cleanup_nodes.sh
-    rm -f /tmp/daos.log /tmp/server.log
+    rm -f /tmp/daos.log /tmp/daos_server.log /tmp/daos_engine.*.log

But that's not a perfect match, because my PR doesn't rename any "server.log" file (only "daos_server*.log" and "daos_control.log").  It seems that this filename is set here:

src/tests/ftest/util/apricot/apricot/test.py
-            os.getenv("D_LOG_FILE", "/tmp/server.log"))[0]
+            os.getenv("D_LOG_FILE", "/tmp/daos_server.log"))[0]

Should we do both of these renames to make everything consistent?

File: utils/config/daos_server.yml (hunk 2 above)
Comment (tanabarr) [reply utils/config/daos_server.yml:307]: can fix this text in https://github.com/daos-stack/daos/pull/4611

**Review by tanabarr — DISMISSED** [review]

**Comment by tanabarr** [comment]
> The only place in CI that I could find is this:
> 
> ci/provisioning/log_cleanup_nodes.sh
> 
> * rm -f /tmp/daos.log /tmp/server.log
> 
> * rm -f /tmp/daos.log /tmp/daos_server.log /tmp/daos_engine.*.log
> 
> But that's not a perfect match, because my PR doesn't rename any "server.log" file (only "daos_server*.log" and "daos_control.log"). It seems that this filename is set here:
> 
> src/tests/ftest/util/apricot/apricot/test.py
> 
> * ```
>          os.getenv("D_LOG_FILE", "/tmp/server.log"))[0]
>   ```
> 
> * ```
>          os.getenv("D_LOG_FILE", "/tmp/daos_server.log"))[0]
>   ```
> 
> Should we do both of these renames to make everything consistent?

I've started renaming the test variables in https://github.com/daos-stack/daos/pull/4611 so I can make that change there

**Comment by johannlombardi** [comment]
Well, it still changes daos_server_unittests.yml which is used by some unit tests AFAIK and won't be collected/cleaned up in our CI. Maybe we should revert this specific change and address it in Tom's follow-on PR.

**Comment by tanabarr** [comment]
moved that change to https://github.com/daos-stack/daos/pull/4611/commits/3af79723cc76aa748487a2f7ce1370da017ea2bc so that the logs gets caught and cleaned up in CI with this change https://github.com/daos-stack/daos/pull/4611/commits/56e5ecc7a0687b7a8387b63ce5c2efd69585cd8d , @phender can you verify that this makes sense please?

**Review by bfaccini — COMMENTED** [review]
@johannlombardi there are a lot more instances of server vs engine usage, mainly in tests and found with '"find . | xargs grep "server.\*\\.log"' from top of daos source tree, should they also be modified ??

@brianjmurrell I believe these additional changes should be harmless as artefacts archival/removal in CI seems to just use the  "\*.log\*" wildcarded pattern.

**Comment by Michael-Hennecke** [comment]
as I mentioned above, the only "oddball" hit this I found is "/tmp/server.log", not "daos_server*.log", which is unrelated to this PR.  

@bfaccini 's grep also matches "server log" texts in comments (blank, not dot), and there are many of those...

**Pull request merged**